synapse/storage/pusher.py

```diff
@@ -135,6 +135,48 @@ def get_all_updated_pushers_txn(txn):
             "get_all_updated_pushers", get_all_updated_pushers_txn
         )
 
+    def get_all_updated_pushers_rows(self, last_id, current_id, limit):
+        """Get all the pushers that have changed between the given tokens.
+
+        Returns:
+            list(tuple): each tuple consists of:
```

Comment:
No, it returns a Deferred, which resolves to that lot.

(I'd prefer it if we even put this on `@defer.inlineCallbacks`, but I won't insist there because it's "obvious". Here though, there is no clue that it actually returns a Deferred, rather than a list.)